wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -279,6 +279,9 @@ public void delete(@ApiParam("The workspace id") @PathParam("id") String id) thr
                                                                                         ConflictException,
                                                                                         ForbiddenException {
         ensureUserIsWorkspaceOwner(id);
+        if (!workspaceManager.getSnapshot(id).isEmpty()) {
```

Comment:
I think no, the main idea of this method name is `get workspace snapshot` which is actually the list of machine snapshots, we also planed to create separate object describing workspace snapshot, thats why this name fits our needs